notebooks/ch15/bias_cv.md

```diff
@@ -0,0 +1,471 @@
+        
+        validation_errors.append(error)
+    
+    #average validation errors
+    return np.mean(validation_errors)
+```
+
+
+```python
+# We train a linear regression classifier for each featurized dataset and perform cross-validation
+# We set fit_intercept=False for our linear regression classifier since 
+# the PolynomialFeatures transformer adds the bias column for us.
+
+cross_validation_errors = [compute_CV_error(LinearRegression(fit_intercept=False), X_train_poly, y_train)
+                     for X_train_poly in X_train_polys]
+```
+
+
+```python
+# HIDDEN
+cv_df = pd.DataFrame({'Validation Error': cross_validation_errors}, index=range(1, 11))
+cv_df.index.name = 'Degree'
+pd.options.display.max_rows = 20
+display(cv_df)
+pd.options.display.max_rows = 7
+```
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>Validation Error</th>
+    </tr>
+    <tr>
+      <th>Degree</th>
+      <th></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>1</th>
+      <td>0.332086</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>0.046725</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>0.047069</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>0.048161</td>
+    </tr>
+    <tr>
+      <th>5</th>
+      <td>0.048352</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>0.048653</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>0.048578</td>
+    </tr>
+    <tr>
+      <th>8</th>
+      <td>0.048877</td>
+    </tr>
+    <tr>
+      <th>9</th>
+      <td>0.051174</td>
+    </tr>
+    <tr>
+      <th>10</th>
+      <td>0.050455</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+We can see that as we use higher degree polynomial features, the validation error decreases and increases again.
+
+
+```python
+# HIDDEN
+plt.figure(figsize=(10, 5))
+
+plt.subplot(121)
+plt.plot(cv_df.index, cv_df['Validation Error'])
+plt.scatter(cv_df.index, cv_df['Validation Error'])
+plt.title('Validation Error vs. Polynomial Degree')
+plt.xlabel('Polynomial Degree')
+plt.ylabel('Validation Error');
+
+plt.subplot(122)
+plt.plot(cv_df.index, cv_df['Validation Error'])
+plt.scatter(cv_df.index, cv_df['Validation Error'])
+plt.ylim(0.046225, 0.05)
+plt.title('Zoomed In')
+plt.xlabel('Polynomial Degree')
+plt.ylabel('Validation Error')
+
+plt.tight_layout();
+```
+
+
+![png](bias_cv_files/bias_cv_22_0.png)
+
+
+Examining the validation errors reveals the most accurate model only used degree 2 polynomial features. Thus, we select the degree 2 polynomial model as our final model and fit it on the all of the training data at once. Then, we compute its error on the test set.
```

Comment:
Maybe "reveals [that] the most accurate model"

Reply:
fixed